### PR TITLE
Correctly calculate average precision, add titles to evaluator plots

### PR DESCRIPTION
**Pull request by jimmyxu-db**
Signed-off-by: Jimmy Xu <jimmy.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

1. Previously, the AP was incorrectly calculated as the mean of the precision values. This has been fixed to use sklearn.metrics.average_precision_score.
2. Added titles to all evaluator plots.

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
3. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Previously, the AP was incorrectly calculated as the mean of the precision values. This has been fixed to use sklearn.metrics.average_precision_score.
Added titles to all evaluator plots.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/evaluation/default_evaluator.py [hunk 1]
```diff
@@ -264,11 +266,11 @@ def gen_line_x_y_label_fn(_y, _y_prob):
 
         def gen_line_x_y_label_fn(_y, _y_prob):
             precision, recall, _thresholds = sk_metrics.precision_recall_curve(_y, _y_prob)
-            ap = np.mean(precision)
+            ap = sk_metrics.average_precision_score(_y, _y_prob)
```
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:269]: Do we need to pass `pos_label` to `average_precision_score`?

Comment (dbczumar) [reply mlflow/models/evaluation/default_evaluator.py]: Yep!

File: mlflow/models/evaluation/default_evaluator.py [hunk 2]
```diff
@@ -260,15 +262,23 @@ def gen_line_x_y_label_fn(_y, _y_prob):
...
+        title = "ROC curve"
+        if pos_label:
+            xlabel = f"False Positive Rate (Positive label: {pos_label})"
+            ylabel = f"True Positive Rate (Positive label: {pos_label})"
     elif curve_type == "pr":
 
         def gen_line_x_y_label_fn(_y, _y_prob):
             precision, recall, _thresholds = sk_metrics.precision_recall_curve(_y, _y_prob)
-            ap = np.mean(precision)
+            ap = sk_metrics.average_precision_score(_y, _y_prob, pos_label)
```
Comment (dbczumar) [inline mlflow/models/evaluation/default_evaluator.py:273]:
```suggestion
            ap = sk_metrics.average_precision_score(_y, _y_prob, pos_label=pos_label)
```

File: mlflow/models/evaluation/default_evaluator.py [hunk 3]
```diff
@@ -240,6 +240,7 @@ def _gen_classifier_curve(
     y_probs,
     labels,
     curve_type,
+    pos_label=None,
```
Comment (dbczumar) [inline mlflow/models/evaluation/default_evaluator.py:243]: Nit: Can we move this one line up so that its colocated with `labels` and remove the `None` default value so that callers always need to specify `pos_label`?

File: mlflow/models/evaluation/default_evaluator.py [hunk 4]
```diff
@@ -260,15 +262,23 @@ def gen_line_x_y_label_fn(_y, _y_prob):
 
         xlabel = "False Positive Rate"
         ylabel = "True Positive Rate"
+        title = "ROC curve"
```
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:265]: Looks like `roc_curve` also has `pos_label`.

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_curve.html

**Review by dbczumar — APPROVED** [review]
LGTM once a couple small comments are addressed. Thanks @jimmyxu-db !

File: mlflow/models/evaluation/default_evaluator.py (hunk 2 above)
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:273]:
```suggestion
            ap = sk_metrics.average_precision_score(_y, _y_prob, pos_label=pos_label)
```

File: mlflow/models/evaluation/default_evaluator.py (hunk 3 above)
Comment (dbczumar) [reply mlflow/models/evaluation/default_evaluator.py:243]: @jimmyxu-db Where are we calling `_gen_classifier_curve` with `pos_label != None`? Right now, looks like we're adding an entirely unused arg. Can we pass a pos label value to all `_gen_classifier_curve()` calls?

File: mlflow/models/evaluation/default_evaluator.py (hunk 2 above)
Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py:273]: Duplicate of https://github.com/mlflow/mlflow/pull/6802/files#r971506041.

File: mlflow/models/evaluation/default_evaluator.py (hunk 4 above)
Comment (dbczumar) [reply mlflow/models/evaluation/default_evaluator.py:265]: Indeed. @jimmyxu-db I'll handle this in my PR

File: mlflow/models/evaluation/default_evaluator.py (hunk 3 above)
Comment (jimmyxu-db) [reply mlflow/models/evaluation/default_evaluator.py:243]: Why should we make pos_label not have a default value? Are you expecting the caller to pass in a pos_label all the time? I would expect that most of the time there is no pos label? For example, would we expect a pos_label if classification is not binary?
